git.go

```diff
@@ -11,11 +11,21 @@ import (
 	"strconv"
 	"strings"
 
+	blocks "github.com/ipfs/go-block-format"
 	cid "github.com/ipfs/go-cid"
 	node "github.com/ipfs/go-ipld-format"
 	mh "github.com/multiformats/go-multihash"
 )
 
+func init() {
+	// Register the block decoder
+	node.DefaultBlockDecoder[cid.GitRaw] = func(b blocks.Block) (node.Node, error) { return DecodeBlock(b) }
```

Comment:
You don't actually need this anonymous function (`DecodeBlock` satisfies the `DecodeBlockFunc` signature).

The CBOR library only uses an anonymous function because *its* `DecodeBlock` function returns a `*cbornode.Node`, not a `node.Node` (to avoid breaking the API).